### PR TITLE
Add zap transform files to source control and update .yml file

### PR DESCRIPTION
**Pull request by TomGuppy**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
-# Start with a minimal pipeline that you can customize to build and deploy your code.
-# Add steps that build, run tests, deploy, and more:
-# https://aka.ms/yaml
-
 trigger:
 - master
 
-pool:
-  vmImage: 'ubuntu-latest'
+pool: UKHO Ubuntu 1804
```
Comment (gregzealley) [inline azure-pipelines.yml:4]: Should this be in the format:
pool:
    vmIMage: "UKHO Ubuntu 1804" ?

It might not : )

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,19 +1,42 @@
...
-  displayName: 'Run a one-line script'
+- task: CmdLine@2
+  inputs:
+    script: 'chmod 777 -R $(Build.ArtifactStagingDirectory)'
+  displayName: "Set chmod permissions (ArtifactStagingDirectory)"
+
+- task: CmdLine@2
+  inputs:
+    script: 'docker run --rm --mount type=bind,source=$(Build.ArtifactStagingDirectory),target=/zap/wrk/ -t owasp/zap2docker-stable zap-full-scan.py -t $(ApplicationUrl) -g gen.conf -r OWASP-Zap-Report.html -x Report.xml || true'
+  displayName: "Run Owasp Zap Full Scan"
```
Comment (gregzealley) [inline azure-pipelines.yml]: Minor - should we use the proper title "OWASP ZAP"

File: azure-pipelines.yml [hunk 3]
```diff
@@ -1,19 +1,42 @@
...
+  inputs:
+    script:  docker run --rm --mount type=bind,source=$(Build.SourcesDirectory)/src,target=/tmp/nunit/ --mount type=bind,source=$(Build.ArtifactStagingDirectory),target=/tmp/report/ mcr.microsoft.com/powershell:ubuntu-18.04 pwsh -File '/tmp/nunit/ZapTransform.ps1'
+  displayName: "Create Nunit Test Report"
+
+- task: PublishTestResults@2
+  inputs:
+    testResultsFormat: 'NUnit'
+    testResultsFiles: 'Converted-OWASP-ZAP-Report.xml'
+    searchFolder: '$(Build.ArtifactStagingDirectory)'
+  displayName: "Publish Owasp Zap Test Report"
```
Comment (gregzealley) [inline azure-pipelines.yml]: Ditto

File: azure-pipelines.yml [hunk 4]
```diff
@@ -1,19 +1,42 @@
...
+
+- task: PublishBuildArtifacts@1
+  inputs:
+    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
+    ArtifactName: 'Owasp Zap HTML Report'
+    publishLocation: 'Container'
+  displayName: "Publish Owasp Zap Report"
```
Comment (gregzealley) [inline azure-pipelines.yml]: Ditto

File: azure-pipelines.yml (hunk 2 above)
Comment (TomGuppy) [reply azure-pipelines.yml]: Updated in lastest push

File: azure-pipelines.yml (hunk 3 above)
Comment (TomGuppy) [reply azure-pipelines.yml]: Updated in lastest push

File: azure-pipelines.yml (hunk 4 above)
Comment (TomGuppy) [reply azure-pipelines.yml]: Updated in lastest push

File: azure-pipelines.yml (hunk 1 above)
Comment (TomGuppy) [reply azure-pipelines.yml:4]: No, that's in the correct for this :)

**Review by gregzealley — APPROVED** [review]

**Pull request merged**